### PR TITLE
Remove SQM depedency from the compiler

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @heejaechang, @dotnet/roslyn-compiler

File: = [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (TyOverby) [inline =]: What is this file and why is it named `=`??

Comment (jaredpar) [inline =]: Probably a command line typo.  I will get rid of it.

**Comment by agocke** [comment]
LGTM

**Comment by TyOverby** [comment]
What behavior does this remove?  What is the behavior of the command line when the flag is used?

**Comment by jaredpar** [comment]
@TyOverby SQM is an abbreviation for the system by which we collect telemetry data.  When present we essentially send usage info back.

**Comment by TyOverby** [comment]
LGTM

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs [hunk 2]
```diff
@@ -1228,7 +1230,6 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
                 PrintFullPaths = printFullPaths,
                 ShouldIncludeErrorEndLocation = errorEndLocation,
                 PreferredUILang = preferredUILang,
-                SqmSessionGuid = sqmSessionGuid,
```
Comment (jasonmalinowski) [inline src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: This is still (admittedly) a public API, does anything break if we change this behavior? I can't imagine so, but it seems odd to me at first.

File: src/VisualStudio/Telemetry/RoslynTelemetrySetup.cs [hunk 3]
```diff
@@ -35,7 +35,7 @@ public void Initialize(IServiceProvider serviceProvider)
                     new VSTelemetryActivityLogger(telemetryService),
                     Logger.GetLogger()));
 
-            Logger.Log(FunctionId.Run_Environment, KeyValueLogMessage.Create(m => m["Version"] = FileVersionInfo.GetVersionInfo(typeof(Sqm).Assembly.Location).FileVersion));
+            Logger.Log(FunctionId.Run_Environment, KeyValueLogMessage.Create(m => m["Version"] = FileVersionInfo.GetVersionInfo(typeof(VisualStudioWorkspace).Assembly.Location).FileVersion));
```
Comment (jasonmalinowski) [inline src/VisualStudio/Telemetry/RoslynTelemetrySetup.cs]: This is probably better off just using `typeof(RoslynTelemetrySetup)`.

**Comment by jasonmalinowski** [comment]
:+1:. Non-critical nitpicks only.

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs (hunk 2 above)
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: Nope.  This value is `internal`.

**Pull request merged**